src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -144,11 +144,11 @@ await resourceNotificationService.PublishUpdateAsync(resource, s =>
 
         async Task SlowDownMonitoringAsync(ResourceEvent lastEvent, CancellationToken cancellationToken)
         {
-            var releaseAfter = DateTime.Now.AddSeconds(30);
+            var releaseAfter = timeProvider.GetLocalNow().AddSeconds(30);
```

Comment:
nit: Use UTC here instead of local? It doesn't seem like time zones need to be involved here.